### ISSUE TITLE
How to install magenta-gpu with a modern version of CUDA ?

### ISSUE THREAD

**Issue by orgicus** [tensorflow/magenta#1667]
I've recently tried to install magenta-gpu however it fails to install due to an error with the networkx package.

I've uploaded a log [here](http://lifesine.eu/so/m.log)

Any tips on running magenta again on a Windows machine would be great.

(Does magenta-gpu work with current versions of CUDA (e.g. 10.1 or 10.2) ?)

Thank you very much for your time,
George

**Comment by cghawthorne** [comment]
You should just be able to use the `magenta` package. Tensorflow no longer requires a separate package for GPU support, so we stopped updating `magenta-gpu`. We tried to update all our docs to just reference the main magenta package, but maybe we missed some? Was there a doc you saw somewhere that recommended installing `magenta-gpu`?

**Comment by orgicus** [comment]
Thank you for the swift reply.

I did actually succeed in installing `magenta`  however

```python
tf.test.is_gpu_available()
```
return `False` which is why I tried to install `magenta-gpu` which used to work (though using an old version of TF).

Is there any way to run magenta with GPU support ?
If so, how ?

**Comment by cghawthorne** [comment]
I'd suggest looking at https://www.tensorflow.org/install/gpu and making sure that you can get a 1.x version of TensorFlow working with GPU first and then try to get it to work with Magenta.

**Comment by orgicus** [comment]
Gotcha, I remember in the past having luck with `tensorflow-gpu==1.10.0` + `magenta==0.3.12` (with TF installed first, then magenta which wouldn't re-install TF since it's close enough to what it required).

I'll close the issue as soon as I can confirm.

Thank you very much for your support: very much appreciated 
❤️

**Comment by orgicus** [comment]
Hi,

Just wanted to add few updates after testing:

- I've managed to have multiple versions of CUDA installed (10.0, 10.1, 10.2) as long as the nVidia  driver was the latest one.
- I've installed tensorflow-gpu 1.15.2 (since magenta requires [1.15 or higher](https://github.com/tensorflow/magenta/blob/master/setup.py#L64)) then installed `magenta` (not `magenta-gpu`, however magenta uninstalled tensorflow-gpu and I was back to the CPU only version.
- I've got both magenta with tensorflow-gpu by installing magenta first, uninstalling tensorflow(cpu), then installing tensorflow-gpu (`is_gpu_available()` returned `True` and I could import `magenta` too)

In the next few days I'll try to train a sketch-rnn model to confirm this works and will close the ticket then.
